src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectMessage.cs

```diff
@@ -40,13 +41,26 @@ public class OpenIdConnectMessage : AuthenticationProtocolMessage
         /// <summary>
         /// Initializes a new instance of the <see cref="OpenIdConnectMessage"/> class.
         /// </summary>
-        public OpenIdConnectMessage() : this(string.Empty) {}
+        public OpenIdConnectMessage() { }
 
         /// <summary>
-        /// Initializes an instance of <see cref="OpenIdConnectMessage"/> class with a specific issuerAddress.
+        /// Initializes an instance of <see cref="OpenIdConnectMessage"/> class with a json string.
         /// </summary>
-        public OpenIdConnectMessage(string issuerAddress) : base(issuerAddress) 
+        public OpenIdConnectMessage(string json)
         {
+            if (string.IsNullOrEmpty(json))
+                LogHelper.LogArgumentNullException("json");
+
+            try
+            {
+                JObject jsonObject = JObject.Parse(json);
+                SetJsonParameters(jsonObject);
```

Comment:
You could choose to write: SetJsonParameters(JObject.Parse(json)); 
Loosing the local variable.